### PR TITLE
[AIRFLOW-185][AIRFLOW-187] PR tool fixes/updates

### PR DESCRIPTION
**Pull request by jlowin**
Dear Airflow Maintainers,

Please accept this PR that addresses the following issues:
- https://issues.apache.org/jira/browse/AIRFLOW-185
- https://issues.apache.org/jira/browse/AIRFLOW-187

185:
- Fix bug with empty versions list

187:
- Add command to setup_git_remotes
- Add guard for running PR tool in a branch with uncommitted changes
- Add option to edit squash commit message
- Message styling for clarity

### CODE REVIEW

File: dev/airflow-pr [hunk 1]
```diff
@@ -152,18 +157,24 @@ def merge_pr(pr_num, target_ref, title, body, pr_repo_desc, local):
...
+        click.style('\nDo you want to squash the PR?', bold=True),
+        textwrap.dedent(
+            """
+            We recommend that you do!
+
+            Squashing will give you an opportunity to edit the new commit message,
+            but the squashed commit will still be attributed to the PR author.
+            GitHub will show that you merged the squash commit but will mark the PR
+            as closed rather than merged (the distinction is
+            puerly cosmetic).
```
Comment (criccomini) [inline dev/airflow-pr]: Nit: puerly typo, and newline seems erroneous here

File: dev/airflow-pr [hunk 2]
```diff
@@ -178,34 +189,43 @@ def merge_pr(pr_num, target_ref, title, body, pr_repo_desc, local):
...
-        primary_author = raw_input(
-            "Enter primary author in the format of \"name <email>\" (or press enter to use %s): " %
-            distinct_authors[0])
+        distinct_authors = sorted(
+            set(commit_authors),
+            key=lambda x: commit_authors.count(x),
+            reverse=True)
+        primary_author = click.prompt(
+            click.style(
+                'Enter the primary author in the format of \"name <email>\"',
```
Comment (criccomini) [inline dev/airflow-pr:204]: Should you keep `(or press enter to use %s)`?

**Comment by criccomini** [comment]
LGTM just a couple of nits

File: dev/airflow-pr (hunk 1 above)
Comment (jlowin) [inline dev/airflow-pr]: Fixed!

File: dev/airflow-pr (hunk 2 above)
Comment (jlowin) [inline dev/airflow-pr:204]: `click.prompt` displays the default in brackets after the prompt, so the new prompt looks like this:

```
Enter the primary author in the format of "name <email>" [jlowin <jlowin@apache.org>]: 
```

**Pull request merged**